src/utilities/params.ts

```diff
@@ -1,13 +1,12 @@
-import { Param, ParamGetSet, ParamExtras, isParamGetSet, isParamGetter } from '@/types'
-import { InvalidRouteParamValueError } from '@/types/invalidRouteParamValueError'
+import { Param, ParamGetSet, ParamExtras, isParamGetSet, isParamGetter, InvalidRouteParamValueError, ParamReturn } from '@/types'
 
 const extras: ParamExtras = {
   invalid: (message?: string) => {
     throw new InvalidRouteParamValueError(message)
   },
 }
 
-const booleanParam: ParamGetSet = {
+const booleanParam: ParamGetSet<boolean> = {
```

Comment:
Adding types to these causes the errors in the utilities. I also coulda sworn I got eslint warnings about always true conditions. 

I'd actually lean toward not adding the types here so they are `ParamGetSet<any>` that way typescript and eslint force us to cover all the possibilities. Otherwise we might miss a typescript error that is possible at runtime.